### ISSUE TITLE
Unlucky sorting of "Possible values are..." error message

### ISSUE THREAD

**Issue by AlexanderLanin** [conan-io/conan#8571]
### Environment Details (include every applicable attribute)
  * Operating System+version: Ubuntu 20.04 in WSL2
  * Compiler+version: clang version 10.0.0-4ubuntu1
  * Conan version: 1.34.0
  * Python version: 3.8.5

### Steps to reproduce
For pretty much any package run `conan install .. -s compiler=clang -s compiler.version=10.0`

### Logs (Executed commands with output) (Include/Attach if Applicable)
```
ERROR: Invalid setting '10.0' is not a valid 'settings.compiler.version' value.
Possible values are ['10', '11', '3.3', '3.4', '3.5', '3.6', '3.7', '3.8', '3.9', '4.0', '5.0', '6.0', '7.0', '7.1', '8', '9']
Read "http://docs.conan.io/en/latest/faq/troubleshooting.html#error-invalid-setting"
```

### Expected behavior
Sort either by increasing number or as in settings.yml (most likely someone sorted it there intentionally).
I though conan is totally out of date since the latest version is 9.
It took me quite a while to notice the 10 and 11 at the beginning.
In settings.yml they are at the end:
```
    clang:
        version: ["3.3", "3.4", "3.5", "3.6", "3.7", "3.8", "3.9", "4.0",
                  "5.0", "6.0", "7.0", "7.1",
                  "8", "9", "10", "11"]
```

**Comment by memsharded** [comment]
Lets see in https://github.com/conan-io/conan/pull/8573, if it is possible to keep the original order of the lists. It is possible that something might break that might be relying on this order, but lets see.

**Comment by memsharded** [comment]
Changed in https://github.com/conan-io/conan/pull/8573, will be released 1.35
